File: 778-reorganize-string/reorganize-string.java
class Solution {
    public String reorganizeString(String s) {
        int[] charCounts = new int[26];
        for (char c : s.toCharArray()) {
            charCounts[c - 'a']++;
        }
        
        // max heap with char, freq
        var pq = new PriorityQueue<int[]>((a, b) -> Integer.compare(b[1], a[1]));
        for (int i = 0; i < 26; i++) {
            if (charCounts[i] > 0) {
                pq.offer(new int[] {i + 'a', charCounts[i]});
            }
        }
        
        StringBuilder str = new StringBuilder();
        int[] block = pq.poll();
        str.append((char) block[0]);
        block[1]--;
        
        while (!pq.isEmpty()) {
            int[] next = pq.poll();
            str.append((char) next[0]);
            next[1]--;
            
            if (block[1] > 0) {
                pq.offer(block); // Missing semicolon fixed
            }
            block = next;
        }
        
        if (block[1] > 0) {
            return "";
        } else {
            return str.toString();
        }
    }
}

File: 778-reorganize-string/reorganize-string.cpp
class Solution {
public:
    string reorganizeString(string s) {
        int n = s.size();
        int charCounts[26] = {};
        for (int i = 0; i < n; i++)
        {
            charCounts[s[i] - 'a']++;
        }
        // max heap with freq, char
        priority_queue<pair<int, char>> pq;
        for (int i = 0; i < 26; i++)
        {
            if (charCounts[i] > 0)
            {
                pq.push({charCounts[i], 'a' + i});
            }
        }
        // String build
        string ans = "";
        pair<int, char> block = pq.top();
        pq.pop();
        ans += block.second;
        block.first--;
        while(!pq.empty()){
            pair<int, char>next = pq.top();
            pq.pop();
            ans += next.second;
            next.first--;
        if (block.first > 0)
        {
            pq.push(block);
        }
        block = next;
    }
    if (block.first > 0)
    {
        return "";
    }
    return ans;
    }
};